llvm/include/llvm/CodeGen/TargetLowering.h

```diff
@@ -354,6 +354,16 @@ class TargetLoweringBase {
     return IsStrictFPEnabled;
   }
 
+  /// Returns true if the target supports static rounding mode for the given
+  /// instruction.
+  virtual bool isStaticRoundingSupportedFor(const Instruction &I) const {
+    return false;
+  }
```

Comment:
This hook doesn't make sense to me, and I don't see how it would be usable

Reply:
This hasn't been addressed. It makes little sense to query this from an IR instruction

Reply:
This is default implementation, RISCV has its own. If you mean `getMachineRoundingMode`, which the first comment refered to, it is removed.

Reply:
This callback is used by target-neutral SelectionDAGBuilder to decide how to lower a gived constrained intrinsic call, to STRICT_FADD, or to FADD_ROUND.

Reply:
Shouldn't DAGBuilder just use the best target-specific implementation of a target-independent operation? The call: 

> call float @llvm.experimental.constrained.fadd.f32(float %x, float %y, metadata !"round.tonearest", metadata !"fpexcept.ignore")

is lowered to STRICT_FADD on X86 because there is no static rounding on this platform and DAG node anyway would be an operation with rounding mode taken from a hardware register. On RISCV, a better representation exist, so SelectionDAGBuilbder can use it.